packages/block-library/src/avatar/hooks.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * WordPress dependencies
+ */
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { store as coreStore, useEntityProp } from '@wordpress/core-data';
+import { __, sprintf } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+
+function getAvatarSizes( sizes ) {
+	const minSize = sizes ? sizes[ 0 ] : 24;
+	const maxSize = sizes ? sizes[ sizes.length - 1 ] : 96;
+	const maxSizeBuffer = Math.floor( maxSize * 2.5 );
+	return {
+		minSize,
+		maxSize: maxSizeBuffer,
+	};
+}
+
+function useAvatar( avatarUrls ) {
+	const { avatarURL: defaultAvatarUrl } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const { __experimentalDiscussionSettings } = getSettings();
+		return __experimentalDiscussionSettings;
+	} );
+	return avatarUrls ? avatarUrls[ avatarUrls.length - 1 ] : defaultAvatarUrl;
+}
```

Comment:
Should we rename the function to `useAvatarUrl` (since that's what it's specifically returning)?

Seems like we're only ever using that one `avatarUrl`, so how about we make it the caller's job to extract it from whatever array?

```suggestion
function useAvatar( avatarUrl ) {
	const { avatarURL: defaultAvatarUrl } = useSelect( ( select ) => {
		const { getSettings } = select( blockEditorStore );
		const { __experimentalDiscussionSettings } = getSettings();
		return __experimentalDiscussionSettings;
	} );
	return avatarUrl ? avatarUrl : defaultAvatarUrl;
}
```

Alternatively, we could even consider _only_ returning the default, and making it the caller's concern to provide whatever overriding avatar:

```suggestion
function useDefaultAvatar() {
	const { avatarURL: defaultAvatarUrl } = useSelect( ( select ) => {
		const { getSettings } = select( blockEditorStore );
		const { __experimentalDiscussionSettings } = getSettings();
		return __experimentalDiscussionSettings;
	} );
	return defaultAvatarUrl;
}
```

To me, this looks like slightly nicer separation of concerns. (Not a big deal though, ofc.)

Reply:
This way, we duplicate the code `avatarUrls ? avatarUrls[ avatarUrls.length - 1 ] : defaultAvatarUrl` on both `useCommentAvatar` and `useUserAvatar` functions.
But, as you mention, we also separate the concerns and create a `useDefaultAvatar` function that does only one thing, following Clean Code standards 😄

Reply:
I'm leaning towards the separation of concerns in this case, even if that means some slight code duplication, but I'll leave the ultimate decision to you 😄

Reply:
Totally agree 🚀 . Now `useDefaultAvatar` only will return the default Avatar Url.